src/System.IO.FileSystem/tests/File/Create_str.cs

```diff
@@ -0,0 +1,231 @@
+            {
+                Assert.False(File.Exists(testFile + "AaAa"));
+                Assert.True(File.Exists(testFile + "AAAA"));
+                Assert.True(File.Exists(testFile + "aAAa"));
+                Assert.Equal(2, Directory.GetFiles(testDir.FullName).Length);
+            }
+            Assert.Throws<DirectoryNotFoundException>(() => File.Create(testFile.ToLowerInvariant()));
+        }
+
+        [Fact]
+        [PlatformSpecific(PlatformID.Windows | PlatformID.OSX)]
+        public void CaseInsensitive()
+        {
+            DirectoryInfo testDir = Directory.CreateDirectory(GetTestFilePath());
+            string testFile = Path.Combine(testDir.FullName, GetTestFileName());
+            File.Create(testFile + "AAAA").Dispose();
+            File.Create(testFile.ToLowerInvariant() + "aAAa").Dispose();
+            Assert.Equal(1, Directory.GetFiles(testDir.FullName).Length);
+        }
+
+        [Fact]
+        [PlatformSpecific(PlatformID.Windows)]
+        public void WindowsWildCharacterPath()
+        {
+            DirectoryInfo testDir = Directory.CreateDirectory(GetTestFilePath());
+            Assert.Throws<ArgumentException>(() => Create(Path.Combine(testDir.FullName, "dls;d", "442349-0", "v443094(*)(+*$#$*", new string(Path.DirectorySeparatorChar, 3))));
+            Assert.Throws<ArgumentException>(() => Create(Path.Combine(testDir.FullName, "*")));
+            Assert.Throws<ArgumentException>(() => Create(Path.Combine(testDir.FullName, "Test*t")));
+            Assert.Throws<ArgumentException>(() => Create(Path.Combine(testDir.FullName, "*Tes*t")));
+        }
+
+        [Fact]
+        [PlatformSpecific(PlatformID.Windows)]
+        public void WindowsWhitespacePath()
+        {
+            Assert.Throws<ArgumentException>(() => Create("         "));
+            Assert.Throws<ArgumentException>(() => Create(" "));
+            Assert.Throws<ArgumentException>(() => Create("\n"));
+            Assert.Throws<ArgumentException>(() => Create(">"));
+            Assert.Throws<ArgumentException>(() => Create("<"));
+            Assert.Throws<ArgumentException>(() => Create("\0"));
+            Assert.Throws<ArgumentException>(() => Create("\t"));
+        }
+
+        [Fact]
+        [PlatformSpecific(PlatformID.AnyUnix)]
+        [ActiveIssue(2205)]
+        public void UnixWhitespacePath()
+        {
+            DirectoryInfo testDir = Directory.CreateDirectory(GetTestFilePath());
+            Assert.Throws<ArgumentException>(() => Create("\0"));
+            using (Create(testDir.FullName + Path.DirectorySeparatorChar + "          "))
+            using (Create(testDir.FullName + Path.DirectorySeparatorChar + " "))
+            using (Create(testDir.FullName + Path.DirectorySeparatorChar + "\n"))
+            using (Create(testDir.FullName + Path.DirectorySeparatorChar + ">"))
+            using (Create(testDir.FullName + Path.DirectorySeparatorChar + ">"))
```

Comment:
Is this a typo?  I'd expect this to fail, not because of the FileSystem implementation, but because this same directory is being created twice and then below we're checking for one that doesn't exist.